metadata.go

```diff
@@ -0,0 +1,45 @@
+package indexer
+
+import (
+	"bytes"
+
+	"github.com/multiformats/go-varint"
+)
+
+// Metadata is data that provides information about retrieving
+// data for an index, from a particular provider.
+type Metadata struct {
+	// ProtocolID defines the protocol used for data retrieval.
+	ProtocolID uint64
```

Comment:
should this be a `uvarint`? or maybe a `multicodec.Code`?

Reply:
It should be a `multicodec.Code` IMO, but that implies that there is at least one value has a documented format. Is it OK to make it of type `multicodeex.Code` without first defining a value in `multicodec`?

Reply:
I'd say yes - there'll be a standardized format soon enough.